sdk/runanywhere-python/runanywhere/events.py

```
@@ -0,0 +1,116 @@
+
+    def emit(self, event: RunAnywhereEvent) -> None:
+        """Emit an event to all listeners; a throwing listener never breaks the emit."""
+        for listener in list(self._listeners):
+            try:
+                listener(event)
+            except Exception:
+                # A misbehaving listener must not disrupt the others. Catch Exception
+                # (NOT BaseException) so control-flow signals — KeyboardInterrupt,
+                # SystemExit, asyncio.CancelledError (a BaseException since 3.8) — still
+                # propagate to the caller instead of being silently swallowed here.
+                pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)